### DISCUSSION TITLE
Intellisense throws errors on c++ includes. (clang 'iostream' file not found)

### DISCUSSION THREAD

**Discussion by wivdvo** [zed-industries/zed#7655] [Feature Requests]
Program complies fine with clang++ even though itellisense gives an error. Also the autocomplete suggests only C headers. (Language in zed is set to c++ and file is .cpp)

**Comment by MuriWolf** [comment]
Having the same error on Linux Mint 21.3

**Comment by hexmarsh** [comment]
You might not have the c++ headers installed on your distro. I'm on fedora, and I had the same issue until I installed libstdc++-devel.

**Reply by MUCCHU** [reply]
Hey thanks, installing `libstdc++-12-dev` on linux mint worked. However I get an warning when I try to include the bist/stdc++ header.
Error message: `In included file: no member named 'fenv_t' in the global namespace; did you mean 'dev_t'?`. Any idea how to fix it?

![image](https://github.com/user-attachments/assets/5092a647-62e4-4ba7-a0b7-e9559fa4c838)

**Comment by owlx56** [comment]
Make sure you have installed the c++ headers.

For missing msvc head files, just install the `C++ build insights` via Visual Studio:

![image](https://github.com/user-attachments/assets/a10075e6-87f1-4c09-a9b9-d4ad0b645534)

For others, try adding compiler executable to `--query-driver` in `clangd/arguments`. Or fixing `clangd/config.yaml` `.clangd`：https://github.com/clangd/clangd/issues/617#issuecomment-1873911283

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
